internal/provider/resource_gitlab_user_gpgkey_test.go

```diff
@@ -0,0 +1,285 @@
+/yG4eD5jp4nGkRfDlXA+ZsIjmChtIRT2DVy/AQDRFeywZIsvuje6DF1AP7qeqbs+
+dZcWp5qRyFu5zodW2gEAgw7OxfYmlQbz7wwKr3IYnY9MyVp+JwSxyAxZ4X+odAM=
+=XSo3
+-----END PGP PUBLIC KEY BLOCK-----`
+var testEd25519GPGPubKeyForCurrentUser string = `-----BEGIN PGP PUBLIC KEY BLOCK-----
+
+mDMEYtzl7hYJKwYBBAHaRw8BAQdAF8RE2x2wE3w/QJAbB8uVv+9HqYQZJGyNZeLt
+eKyyh2i0IFRlcnJhZm9ybSA8dGVycmFmb3JtQGdpdGxhYi5jb20+iJAEExYIADgW
+IQShzcoy44uIRzOHNnGXiWgn/RdiBQUCYtzl7gIbAwULCQgHAgYVCgkICwIEFgID
+AQIeAQIXgAAKCRCXiWgn/RdiBUDpAQDpcNH5rxVlJS4ATpkFPxbyMuZBdr6eaw2G
+3o8ZPykfbAEAt62eRBhyep+rjPyAHW8YyX4+e+SdCZw/KH0NDbZCGAW4MwRi3OXu
+FgkrBgEEAdpHDwEBB0DE69f1NA/05px8W0ZUkIl0kheMosQ3+UFbt/Jba0DQiojv
+BBgWCAAgFiEEoc3KMuOLiEczhzZxl4loJ/0XYgUFAmLc5e4CGwIAgQkQl4loJ/0X
+YgV2IAQZFggAHRYhBCB2JV+gsGWeld0PcwehRlcCKxpcBQJi3OXuAAoJEAehRlcC
+KxpcSmcBALG/ZILOrVHratA+cZchDQEtOM2rymXdw6AgnllhMj48AQCjWmESbs+C
+GcNNwjo6hAu59BiCvU5+W2of9fpSxBM5CJQ4AQDjyZwDcf6kMu5+bYY9aNcz9skX
+BEMqhn2i2EtNNfH6eAEA1m21vnE8pseBCYHtl9/XJGkB5JH0gUqqRrCf5CAqsgk=
+=/bV6
+-----END PGP PUBLIC KEY BLOCK-----`
+var testEd25519GPGPubKeyWithTrailingNewline string = `-----BEGIN PGP PUBLIC KEY BLOCK-----
+
+mDMEYtzSYhYJKwYBBAHaRw8BAQdAaSePzlrqUWT/hBRqO/oIdfBPh5m57cPwpmjL
+dgUJGcW0IFRlcnJhZm9ybSA8dGVycmFmb3JtQGdpdGxhYi5jb20+iJAEExYIADgW
+IQTFohelsWBqQdLTMuS4BhhIqbpAJQUCYtzSYgIbAwULCQgHAgYVCgkICwIEFgID
+AQIeAQIXgAAKCRC4BhhIqbpAJX8hAP9XCLVIMzO8pleoU82XV+yEVxgKA44uSvat
+zcsIlQrUpwEAkFllbHnquiMxOr+UTXFXIj2L48V2oysbwbhadfIeGwO4MwRi3NJi
+FgkrBgEEAdpHDwEBB0AzJRFF4Ufc8MFl42TDhkMNvMYxVdHogLyOXwK5zoQIo4jv
+BBgWCAAgFiEExaIXpbFgakHS0zLkuAYYSKm6QCUFAmLc0mICGwIAgQkQuAYYSKm6
+QCV2IAQZFggAHRYhBKkMHhbuzwzdKeMiMw7wawPgy8RgBQJi3NJiAAoJEA7wawPg
+y8RgobsA/1uo4Z1ybThL//tNXFeV8J4vr5Kj+1mUO92v4E8oMhL0AQDsWH2H6kn2
+yXEq4d7IEMN2bZOgnIK8q9wcQm7ouYXkDZ2rAP96iz2c2HYWouFZXf0vqzrgJWVB
+ubvgn0HwJkdP6VIUKwEArIF90cFLtKmephxMlwp7h+djEWpVIQ/T31pLYSAH4QA=
+=5+sB
+-----END PGP PUBLIC KEY BLOCK-----
+`
+
+func TestAccGitlabUserGPGKey_basic(t *testing.T) {
+	var key gitlab.GPGKey
+	testUser := testAccCreateUsers(t, 1)[0]
+
+	resource.ParallelTest(t, resource.TestCase{
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckGitlabUserGPGKeyDestroy,
+		Steps: []resource.TestStep{
+			// Create a user + gpgkey
+			{
+				Config: testAccGitlabUserGPGKeyConfig(testUser.ID, testEd25519GPGPubKey),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabUserGPGKeyExists("gitlab_user_gpgkey.foo_key", &key),
+					testAccCheckGitlabUserGPGKeyAttributes(&key, &testAccGitlabUserGPGKeyExpectedAttributes{
+						Key:   testEd25519GPGPubKey,
+					}),
+				),
+			},
+			// Update the key
+			{
+				Config: testAccGitlabUserGPGKeyConfig(testUser.ID, updatedEd25519GPGPubKey),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabUserGPGKeyExists("gitlab_user_gpgkey.foo_key", &key),
+					testAccCheckGitlabUserGPGKeyAttributes(&key, &testAccGitlabUserGPGKeyExpectedAttributes{
+						Key:       updatedEd25519GPGPubKey,
+					}),
+				),
+			},
```

Comment:
Can you add a test case for when someone switches from the current user to another arbitrary user gpg key (and maybe vice-versa) ?

Reply:
@yan12125 I've noticed your 👍 , but haven't found the test. Did you forget to push it?

Reply:
Sorry for confusion. I like the idea but had no time to add that yesterday. Now a test is expanded :)